app/controllers/protocols_controller.rb

```diff
@@ -60,8 +60,13 @@ class ProtocolsController < ApplicationController
     copy_to_repository
     copy_to_repository_modal
   )
-  before_action :check_import_permissions, only: [:import]
-  before_action :check_export_permissions, only: [:export]
+  before_action :check_import_permissions, only: :import
+  before_action :check_export_permissions, only: :export
+
+  before_action :check_user_role,
+                only: %i(
+                  import protocolsio_import_create protocolsio_import_save
```

Comment:
Then just remove `import` from here. And maybe rename before_action to `check_protocolsio_import_permissions`